src/Hedgehog/Outcome.fs

```diff
@@ -7,14 +7,19 @@ type Outcome<'a> =
 
 module Outcome =
 
-    let cata result failure discard success =
-        match result with
+    let private cata
+        (outcome : Outcome<'a>)
+        (failure : unit -> 'b)
+        (discard : unit -> 'b)
+        (success :   'a -> 'b)
+            : 'b =
+        match outcome with
         | Failure ->
-            failure()
+            failure ()
         | Discard ->
-            discard()
-        | Success(x) ->
-            success(x)
+            discard ()
+        | Success x ->
+            success x
 
     [<CompiledName("Map")>]
```

Comment:
We might not need the `CompiledName` attributes here, since we have `Hedgehog.Linq`.

Reply:
:thinking: